modules/cluster/cloud-config.yml

```diff
@@ -1,112 +1,111 @@
+            },
+            "log_stream_name": "{instance_id}/untitled-logs"
+        }
+    }
 
-  - path: "/etc/ecs/ecs.config"
-    permissions: "0744"
-    owner: "root"
-    content: |
-      ECS_CLUSTER=${ecs_cluster_name}
-      ECS_LOG_LEVEL=${ecs_log_level}
-      ECS_ENABLE_CONTAINER_METADATA=true
-      ECS_ENABLE_TASK_IAM_ROLE=true
-      ECS_ENABLE_TASK_IAM_ROLE_NETWORK_HOST=true
-      ECS_AVAILABLE_LOGGING_DRIVERS=["awslogs"]
+- path: "/etc/ecs/ecs.config"
+  permissions: "0744"
+  owner: "root"
+  content: |
+    ECS_CLUSTER=${ecs_cluster_name}
+    ECS_LOG_LEVEL=${ecs_log_level}
+    ECS_ENABLE_CONTAINER_METADATA=true
+    ECS_ENABLE_TASK_IAM_ROLE=true
+    ECS_ENABLE_TASK_IAM_ROLE_NETWORK_HOST=true
+    ECS_AVAILABLE_LOGGING_DRIVERS=["awslogs"]
 
-  - path: "/usr/local/scripts/cloudformation-signal.sh"
-    permissions: "0744"
-    owner: "root"
-    content: |
-      #! /bin/bash
-
-      set -euo pipefail
-
-      function await_process() {
-        echo -n "Waiting for $1..."
-        while ! pgrep -f "$1" > /dev/null; do
-          sleep 1
-        done
-        echo "Done!"
-      }
-      await_process "/usr/libexec/amazon-ecs-init start"
+- path: "/usr/local/scripts/cloudformation-signal.sh"
+  permissions: "0744"
+  owner: "root"
+  content: |
+    #! /bin/bash
+    set -uo pipefail
+    INSTANCE_ID=$(curl http://169.254.169.254/latest/meta-data/instance-id)
+    echo -n "Waiting for Instance $INSTANCE_ID to be ACTIVE in cluster ${ecs_cluster_name}"
+    function await_process() {
+      CHECK_ACTIVE=$(aws --region ${region} ecs list-container-instances --cluster ${ecs_cluster_name} --status ACTIVE --filter ec2InstanceId==$INSTANCE_ID | grep 'arn:aws:ecs')
+      while [ -z $CHECK_ACTIVE ]; do
+        sleep 1
+        CHECK_ACTIVE=$(aws --region ${region} ecs list-container-instances --cluster ${ecs_cluster_name} --status ACTIVE --filter ec2InstanceId==$INSTANCE_ID | grep 'arn:aws:ecs')
+      done
+      echo "Done!"
+      /opt/aws/bin/cfn-signal -e $? --stack ${stack_name} --resource AutoScalingGroup --region ${region}
+    }
+    await_process
 
 runcmd:
-  - |
-    yum install -y https://amazon-ssm-${region}.s3.amazonaws.com/latest/linux_amd64/amazon-ssm-agent.rpm
-  - |
-    mkdir -p /tmp/cloudwatchagent && cd /tmp/cloudwatchagent
-    curl -O https://s3.amazonaws.com/amazoncloudwatch-agent/linux/amd64/latest/AmazonCloudWatchAgent.zip
-    unzip AmazonCloudWatchAgent.zip && ./install.sh
-    rm -rf /tmp/cloudwatchagent
-    start amazon-cloudwatch-agent
-  - |
-    start ecs
-    /usr/local/scripts/cloudformation-signal.sh
-    /opt/aws/bin/cfn-signal -e $? --stack ${stack_name} --resource AutoScalingGroup --region ${region}
+- |
+  yum install -y https://s3.dualstack.${region}.amazonaws.com/amazon-ssm-${region}/latest/linux_amd64/amazon-ssm-agent.rpm
+  curl -O https://bootstrap.pypa.io/get-pip.py
+  python get-pip.py --user
+  export PATH=~/.local/bin:$PATH
+  pip install awscli --upgrade --user
```

Comment:
Assuming `--user` means "install just for the user", is there any reason it can't just be a global install?

Reply:
Not 100% sure - instructions from AWS followed.  However since it does what it needs ands follows a principal of least privilege by not installing it for all users I am happy with the way it is